modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/L0PeerDiscovery.scala

```diff
@@ -39,6 +39,10 @@ sealed abstract class L0PeerDiscovery[F[_]: Sync: Random] private (
 
   val logger = Slf4jLogger.getLogger[F]
 
+  // TODO Refactor to eliminate race condition?
+  //  There is a small window where after calling getPeers and before calling getPeer
+  //  the peers could be reset (setPeers) and so getPeer might return None.
+  //  We can keep the result of `getPeers` and avoid calling `getPeer` later
```

Comment:
Instead of TODO in the code, please create a ticket in jira

Reply:
Created https://constellationnetwork.atlassian.net/browse/PROT-350

cc: @ryle-goehausen